src/core/api/api_document.pl

```diff
@@ -183,18 +181,24 @@
                          empty_assoc(Captures),
                          nb_thread_var_init(Captures, Captures_Var),
                          (   Full_Replace = true
-                         ->  replace_existing_graph(Graph_Type, Transaction, Stream, Captures_Var),
-                             Ids = []
-                         ;   ensure_transaction_has_builder(Graph_Type, Transaction),
-                             findall(Id,
-                                     (   json_read_dict_list_stream(Stream, Document),
-                                         nb_thread_var({Graph_Type,Transaction,Document,Id}/[State,Captures_Out]>>(api_insert_document_(Graph_Type, Transaction, Document, State, Id, Captures_Out)),
-                                                       Captures_Var)),
-                                     Ids),
-
-                             die_if(nonground_captures(Captures_Var, Nonground),
-                                    error(not_all_captures_found(Nonground), _)),
-                             die_if(has_duplicates(Ids, Duplicates), error(same_ids_in_one_transaction(Duplicates), _)))),
+                         ->  api_nuke_documents_(Graph_Type, Transaction)
+                         ;   true
+                         ),
+                         prepare_full_replace(Full_Replace, Graph_Type, Stream, Transaction, Tail_Stream),
+                         ensure_transaction_has_builder(Graph_Type, Transaction),
```

Comment:
you should ensure the builder before you prepare the full replace as that already potentially uses the builder.